build/xtask/src/dist.rs

```diff
@@ -660,21 +660,17 @@ fn translate_elf_to_other_formats(
 }
 
 fn write_gdb_script(cfg: &PackageConfig, image_name: &str) -> Result<()> {
-    // Humility doesn't know about images right now. The gdb symbol file
-    // paths all assume a flat layout with everything in dist. For now,
-    // match what humility expects. If a build file ever contains multiple
-    // images this will need to be fixed!
     let mut gdb_script = File::create(cfg.img_file("script.gdb", image_name))?;
     writeln!(
         gdb_script,
         "add-symbol-file {}",
-        cfg.dist_file("kernel").to_slash().unwrap()
+        cfg.img_file("kernel", image_name).to_slash().unwrap()
```

Comment:
The motivation for this change makes sense, but what was the actual change in behavior for using the gdb script?  I have not had any problems debugging archives generated after https://github.com/rivosinc/hubris/commit/0b29eb59b5910c0279bfb4c16913565711005c34.

Reply:
I've attached debug sessions below (from before/after gdb script change) that show an attempt to debug jefe's main() function.

Before the change, gdb fails to execute script.gdb as it fails to 'add-symbol-file' for jefe (due to 'No such file or directory').  This is due to the fact that 0b29eb5 moves build artifacts around (there is no longer a 'dist/jefe' as we now only have 'dist/{image_name}/jefe').

After the change, gdb executes the 'add-symbol-file' commands in 'script.gdb' successfully and we are able to set breakpoints in the jefe task.

Before change:
> $ RUST_GDB=riscv64-unknown-elf-gdb rust-gdb -x target/demo-rv64-qemu-virt-supervisor/dist/default/script.gdb -ex "target remote :1234"
> GNU gdb (GDB) 12.0.50.20220419-git
> Copyright (C) 2022 Free Software Foundation, Inc.
> [...]
> add symbol table from file "target/demo-rv64-qemu-virt-supervisor/dist/kernel"
> add symbol table from file "target/demo-rv64-qemu-virt-supervisor/dist/jefe"
> target/demo-rv64-qemu-virt-supervisor/dist/default/script.gdb:2: Error in sourced command file:
> target/demo-rv64-qemu-virt-supervisor/dist/jefe: No such file or directory.
> Remote debugging using :1234
> warning: No executable has been specified and target does not support
> determining executable automatically.  Try using the "file" command.
> 0x0000000080000000 in ?? ()
> (gdb) break task_jefe::main
> Function "task_jefe::main" not defined.
> Make breakpoint pending on future shared library load? (y or [n]) n
> (gdb) 

After change:
> $ RUST_GDB=riscv64-unknown-elf-gdb rust-gdb -x target/demo-rv64-qemu-virt-supervisor/dist/default/script.gdb -ex "target remote :1234"
> GNU gdb (GDB) 12.0.50.20220419-git
> Copyright (C) 2022 Free Software Foundation, Inc.
> [...]
> add symbol table from file "target/demo-rv64-qemu-virt-supervisor/dist/default/kernel"
> add symbol table from file "target/demo-rv64-qemu-virt-supervisor/dist/default/jefe"
> add symbol table from file "target/demo-rv64-qemu-virt-supervisor/dist/default/pong"
> add symbol table from file "target/demo-rv64-qemu-virt-supervisor/dist/default/ping"
> add symbol table from file "target/demo-rv64-qemu-virt-supervisor/dist/default/idle"
> Remote debugging using :1234
> warning: No executable has been specified and target does not support
> determining executable automatically.  Try using the "file" command.
> 0x0000000080000000 in ?? ()
> (gdb) break task_jefe::main
> Breakpoint 1 at 0x90008a34: file task/jefe/src/main.rs, line 121.
> (gdb) c
> Continuing.
> 
> Thread 1 hit Breakpoint 1, task_jefe::main () at task/jefe/src/main.rs:121
> 121	    sys_log!("viva el jefe");
> (gdb)

Reply:
Hm, this is strange.  I just tried it without this PR and works fine...
Did anyone else experience this before?

Reply:
I see the same behavior for `tests-hifive-inventor` no rivos/main branch. Its fixed by picking this change.

Before:
```
add symbol table from file "target/tests-hifive-inventor/dist/runner"
target/tests-hifive-inventor/dist/default/script.gdb:2: Error in sourced command file:
```

After:
```
add symbol table from file "target/tests-hifive-inventor/dist/default/kernel"
add symbol table from file "target/tests-hifive-inventor/dist/default/runner"
add symbol table from file "target/tests-hifive-inventor/dist/default/suite"
add symbol table from file "target/tests-hifive-inventor/dist/default/assist"
add symbol table from file "target/tests-hifive-inventor/dist/default/idol"
add symbol table from file "target/tests-hifive-inventor/dist/default/idle"
```

BTW, this change will have to be cherry-picked to rivos/main branch as well.